src/engine/runner/shared/docker-compose.parser.ts

```diff
@@ -0,0 +1,37 @@
+// import { load, dump } from "js-yaml";
+import { readFileSync, writeFileSync } from "fs";
+
+export class DockerComposeParser {
+  private document: string[];
+
+  constructor(private path: string) {
+    const document = readFileSync(path, "utf8");
+    this.document = document.split("\n");
+  }
+
+  removeService(serviceKey: string) {
+    this.removeSection(serviceKey);
+    this.removeDependsOn(serviceKey);
+    this.persist();
+  }
+
+  private removeSection(serviceKey: string) {
+    const serviceIndex = this.document.findIndex((value) => value.includes(`${serviceKey}:`));
```

Comment:
Are you sure that we want to edit this yaml this way? I strongly disadvice you this approach since it may cause hard to predict issues.

Reply:
Do you know how we can make it "smarter"? For now, it works. I assume that it can break when somebody changes/adds something in the future. But it's hard to predict right now.

Reply:
Simplest way is to parse yaml to json and then when you are done with processing convert it back to yaml.
Those 2 formats are interchangable between each other.

Reply:
I couldn't find a solution to properly output JSON to YAML, with keeping the correct usage of anchors. Because of some JSON limitations, the libraries need to make a workaround to generate correct YAML files. So in the end there are too many weird changes. That's why I wrote this solution. 
Anyway, If you find a solution for it, feel free to add a PR with it. It's just a first version, so we can easily improve it in the future.